pkgs/build-support/gradle/compress-deps-json.py

```diff
@@ -0,0 +1,94 @@
+import json
```

Comment:
Any particular reason why this is not implemented in Nix? I Skimming over it, it looks like it could. Or maybe [`jq`](https://github.com/jqlang/jq#jq)? Probably this should not block this PR, I'm still curious, and could also re-implement it for you.

Reply:
it's not implemented in Nix because

1. it's a fairly complex piece of logic (which used to be even more complex)
2. it runs in a shell script
3. Nix doesn't have JSON pretty printing

If you can produce identical JSON files in Nix I'd be willing to replace the code with Nix.

To get the original files, you can modify Gradle's fetch-deps.nix to write the processed data to disk instead of passing it to mitm-cache.fetch